tests/Type/CollectionOfObjectsIntegrationTest.php

```diff
@@ -107,99 +107,109 @@ public function testRemove(Foo $foo): void
 	}
 
 	/**
-	 * @dataProvider fooDataProvider
-	 *
-	 * @param \Consistence\Sentry\SymfonyBundle\Type\Foo $foo
+	 * @return mixed[][]|\Generator
 	 */
-	public function testSetInvalidCollectionType(Foo $foo): void
+	public function invalidTypeDataProvider(): Generator
```

Comment:
I do not think that this DP and the connected ones make much sense like this - in this method it cannot be seen related to what the invalid method should be and also the `testSetEventDatesWithInvalidType()` does not really match the other methods here.

But I think we can just make one DP `invalidArgumentTypeDataProvider` with `callMethodCallback` and then the expected parameters and it can be used for all of these plus also it is extendable for further cases.

I hope I have not overlooked something :) If it is ok, it should be also applied in the other similar cases of course.

Reply:
Nice! This is deduplication proper :)

Changed here and in `CollectionOfStringsIntegrationTest`, `ObjectIntegrationTest`, `StringIntegrationTest`.

Reply:
I think we do not need `invalidValue` since it is only passed back into the DP and decided there what to do with it?

Reply:
Ok, I think it should still be passed to the DP as `expectedInvalidValue`, if that's what you meant :)

Reply:
I overlooked that, but yes, it has to be there :) (and I think like this it is much better).

But the same change should be applied to the other places as well please?

Reply:
Should be fixed now, sorry for having missed that.